src/tools/imageSrcset.js

```diff
@@ -0,0 +1,40 @@
+export default class {
+    getSrcset(url, sizes) {
+        // Regex to test size string is of the form 123x123 or 100w
+        const sizeRegex = /(^\d+w$)|(^(\d+?)x(\d+?)$)/;
+        // Regex to test to that srcset descriptor is of the form 1x 1.5x 2x OR 123w
+        const descriptorRegex = /(^\d+w$)|(^([0-9](\.[0-9]+)?)x)$/;
+
+        let srcsets = {};
+
+        if (!sizes) {
+            // Use default srcsets if none are provided
+            srcsets = {
+                '80w': '80w',
+                '160w': '160w',
+                '320w': '320w',
+                '640w': '640w',
+                '960w': '960w',
+                '1280w': '1280w',
+                '1920w': '1920w',
+                '2560w': '2560w',
+            };
+        } else if ((sizes === Object(sizes)) && !Object.keys(sizes).some(descriptor =>
```

Comment:
`sizes === Object(sizes)` does this mean we are check if it is of type object ?

Reply:
Correct; I'm trying to catch cases where people give bad input and provide them with an error.

Check these test cases: https://github.com/bigcommerce/stencil-utils/pull/106/files#diff-2f4a919860db4dbc3b9809590bce50c1R81